corehq/apps/sms/api.py

```diff
@@ -54,6 +55,7 @@ def add_msg_tags(msg, metadata):
         msg.reminder_id = metadata.reminder_id
         msg.chat_user_id = metadata.chat_user_id
         msg.ignore_opt_out = metadata.ignore_opt_out
+        msg.location_id = metadata.location_id
```

Comment:
We should do a check here first for:

    if metadata.location_id:
        msg.location_id = metadata.location_id

otherwise it can potentially be overwritten by subsequent calls to add_msg_tags from other handlers.  Probably a good idea to just do this for all these fields too.  We can make it something like:

    fields = ('workflow', 'xforms_session_couch_id', 'reminder_id', 'chat_user_id', 'ignore_opt_out', 'location_id')
    for field in fields:
        value = getattr(metadata, field)
        if value:
             setattr(msg, field, value)